graylog2-server/src/main/java/org/graylog2/indexer/IndexSetRegistry.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.indexer;
+
+import org.graylog2.indexer.indices.TooManyAliasesException;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+public interface IndexSetRegistry {
+    /**
+     * Returns a list of all {@link IndexSet} instances.
+     *
+     * @return list of index sets
+     */
+    List<IndexSet> getAllIndexSets();
+
+    /**
+     * Executes the given consumer function for every {@link IndexSet}.
+     *
+     * @param action the consumer function
+     */
+    void forEach(Consumer<IndexSet> action);
```

Comment:
This seems to be not really necessary, as we can simply call `getAllIndexSets().forEach()`.